app/web/Dockerfile

```diff
@@ -38,4 +38,4 @@ ENV PORT 3000
 # Uncomment the following line in case you want to disable telemetry.
 ENV NEXT_TELEMETRY_DISABLED 1
 
-CMD ["node_modules/.bin/next", "start"]
+CMD ["yarn", "build", "&&", "yarn", "install", "--production", "--ignore-scripts", "--prefer-offline", "&&", "node_modules/.bin/next", "start"]
```

Comment:
Could we not skip `yarn build` here since it's already been run in a previous image layer?

Reply:
We need to build at run-time because of how next handles env vars.
Our choices are either to have 3 separate containers (dev, staging, prod) and run `yarn build` at build time, or have one container and build at run-time.
The first build further up the file seems to speed up subsequent builds so it makes starting the container faster.

Reply:
Sounds like that first build must be populating the cache.
Curious if previous build's cache would speed up the current build. If yes, would it be faster to then restore the cache rather than running `yarn build` twice, as that step in itself seems to take ~5-6 mins